crates/orchestrator/src/spawner.rs

```diff
@@ -174,14 +176,31 @@ where
         )
     })?;
 
-    let ws_uri = format!("ws://{}:{}", LOCALHOST, node.rpc_port.0);
-    let prometheus_uri = format!("http://{}:{}/metrics", LOCALHOST, node.prometheus_port.0);
+    // Create port-forward iff we are not in CI
+    let (rpc_port_external, prometheus_port_external) = if !running_in_ci() {
+        let ports = futures::future::try_join_all(vec![
+            running_node.create_port_forward(node.ws_port.0, RPC_PORT),
+            running_node.create_port_forward(node.prometheus_port.0, PROMETHEUS_PORT),
+        ])
+        .await?;
+
+        (
+            ports[0].unwrap_or(node.ws_port.0),
+            ports[0].unwrap_or(node.prometheus_port.0),
+        )
+    } else {
+        (node.ws_port.0, node.prometheus_port.0)
+    };
```

Comment:
I think the "default" is a bit redundant (```node.ws_port.0``` and  ```node.prometheus_port.0```), maybe we could refacto to have the default initialized and then if not running in CI, creating the port forward, also the node.prometheus_port.0 should be assigne port[1] line 189:

```suggestion

    let (mut rpc_port_external, mut prometheus_port_external) =
        (node.ws_port.0, node.prometheus_port.0);

    if !running_in_ci() {
        let ports = futures::future::try_join_all([
            running_node.create_port_forward(node.ws_port.0, RPC_PORT),
            running_node.create_port_forward(node.prometheus_port.0, PROMETHEUS_PORT),
        ])
        .await?;

        (rpc_port_external, prometheus_port_external) = (ports[0].unwrap(), ports[1].unwrap());
    }
    ```

Reply:
Sounds good, I will change it. 👍